### PR TITLE
changes to build zig-bootstrap aarch64-windows

### PR DESCRIPTION
**Pull request by marler8997**
These are changes I made in order to build zig with the zig-bootstrap project for `aarch64-windows-gnu`

### CODE REVIEW

File: src/windows_sdk.cpp [hunk 1]
```diff
@@ -116,6 +119,10 @@ static ZigFindWindowsSdkError find_msvc_lib_dir(ZigWindowsSDKPrivate *priv) {
             case NativeArchArm:
                 out_append_ptr += sprintf(out_append_ptr, "arm\\");
                 break;
+            case NativeArchAarch64:
+                // TODO: is this right?
+                out_append_ptr += sprintf(out_append_ptr, "aarch64\\");
```
Comment (andrewrk) [inline src/windows_sdk.cpp:124]: Hmm do we know anyone who is running windows on aarch64?

Comment (snickler) [inline src/windows_sdk.cpp:124]:
```suggestion
                out_append_ptr += sprintf(out_append_ptr, "arm64\\");
```

File: src/windows_sdk.cpp [hunk 2]
```diff
@@ -161,6 +168,10 @@ com_done:;
     case NativeArchArm:
         tmp_buf_append_ptr += sprintf(tmp_buf_append_ptr, "arm\\");
         break;
+    case NativeArchAarch64:
+        // TODO: is this right?
+        tmp_buf_append_ptr += sprintf(tmp_buf_append_ptr, "aarch64\\");
```
Comment (snickler) [inline src/windows_sdk.cpp:173]:
```suggestion
        tmp_buf_append_ptr += sprintf(tmp_buf_append_ptr, "arm64\\");
```

File: src/windows_sdk.cpp [hunk 3]
```diff
@@ -23,16 +23,19 @@ enum NativeArch {
...
 static const NativeArch native_arch = NativeArchArm;
-#endif
-#if defined(_M_IX86) || defined(__i386__)
+#elif defined(_M_IX86) || defined(__i386__)
 static const NativeArch native_arch = NativeArchi386;
-#endif
-#if defined(_M_X64) || defined(__x86_64__)
+#elif defined(_M_X64) || defined(__x86_64__)
 static const NativeArch native_arch = NativeArchx86_64;
+#elif defined(__aarch64__)
```
Comment (snickler) [inline src/windows_sdk.cpp:35]:
```suggestion
#elif defined(_M_ARM64) || defined(__aarch64__)
```

**Comment by marler8997** [comment]
@snickler I've committed your suggestions, are you confirming the new `_M_ARM64` define and the paths to `arm64` are correct?  Is this on a machine you have or from documentation somewhere?

**Comment by snickler** [comment]
> @snickler I've committed your suggestions, are you confirming the new `_M_ARM64` define and the paths to `arm64` are correct? Is this on a machine you have or from documentation somewhere?

Yes, the define and paths are confirmed - both from documentation and from having the MSVC ARM64 libraries installed with my instances of Visual Studio. 

https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-160
https://devblogs.microsoft.com/cppblog/windows-arm64-support-for-cmake-projects-in-visual-studio/

Since these builds are using the Microsoft hosted agents, the required MSVC ARM64 components are preinstalled.

https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md

**Comment by andrewrk** [comment]
Thanks for dropping by, @snickler!

**Comment by snickler** [comment]
No problem!

**Pull request merged**